bin/varnishncsa/varnishncsa.c

```diff
@@ -884,7 +902,7 @@ frag_fields(int force, const char *b, const char *e, ...)
 		assert(p != NULL && q != NULL);
 		if (p >= e || q <= p)
 			continue;
-		if (frag->gen != CTX.gen || force) {
+		if (process_format_policy(fp, frag->gen != CTX.gen)) {
```

Comment:
It looks like so far [1] that the "first" criterion is always a generation check. Maybe pass `frag` to `process_format_policy()` and derive first inside the function.

[1] commit "varnishncsa: Introduce process_format_policy()"

Reply:
Forgot to check, so I just did and it looks like the generation is always the criterion, so please pass `frag` instead of a boolean.